src/main/java/at/ac/tuwien/kr/alpha/common/depgraph/Node.java

```diff
@@ -0,0 +1,134 @@
+package at.ac.tuwien.kr.alpha.common.depgraph;
+
+import at.ac.tuwien.kr.alpha.common.Predicate;
+
+/**
+ * A node in a dependency graph. One node references exactly one predicate. This means that all rule heads deriving the same predicate will be condensed into
+ * the same graph node. In some cases this results in more "conservative" results in stratification analysis, where some rules will not be evaluated up-front,
+ * although that would be possible.
+ * 
+ * Note that constraints are represented by one dummy predicate (named "constr_{num}"). Each constraint node has a negative edge to itself to express the
+ * notation of a constraint ":- a, b." as "x :- a, b, not x.".
+ *
+ */
+// TODO ensure immutability
+public class Node {
+
+	public static class NodeInfo {
+
+		private Node dfsPredecessor;
+		private int dfsDiscoveryTime;
+		private int dfsFinishTime;
+		private int componentId;
```

Comment:
Except for the `componentId` no other member value seems to be accessed. Can this class be replaced with just an `int componentId`?